packages/creator-hub/main/src/modules/default-editor-settings.ts

```diff
@@ -0,0 +1,126 @@
+import path from 'path';
+import fs from 'fs/promises';
+import log from 'electron-log/main';
+import { type EditorConfig } from '/shared/types/config';
+
+import { getConfig } from './config';
+
+export enum EditorType {
+  VSCode = 'Visual Studio Code',
+  Cursor = 'Cursor',
+}
+
+const EDITOR_PATHS = {
+  win32: {
+    [EditorType.VSCode]: (username: string) => [
+      `C:\\Users\\${username}\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe`,
+      'C:\\Program Files\\Microsoft VS Code\\Code.exe',
+    ],
+    [EditorType.Cursor]: (username: string) => [
+      `C:\\Users\\${username}\\AppData\\Local\\Programs\\Cursor\\Cursor.exe`,
+      'C:\\Program Files\\Cursor\\Cursor.exe',
+    ],
+  },
+  darwin: {
```

Comment:
wdyt about using something like: `system_profiler -detailLevel basic -json SPApplicationsDataType` (osx only, but windows should have something similar) for getting the apps installed apps on the os and then filtering for the ones we support? maybe the user uses a portable version of the progeam or has it installed in another drive other that C:\\, etc